packages/rocketchat-message-pin/server/pinMessage.js

```diff
@@ -42,16 +42,20 @@ Meteor.methods({
 		originalMessage = RocketChat.callbacks.run('beforeSaveMessage', originalMessage);
 		RocketChat.models.Messages.setPinnedByIdAndUserId(originalMessage._id, originalMessage.pinnedBy, originalMessage.pinned);
 
-		return RocketChat.models.Messages.createWithTypeRoomIdMessageAndUser('message_pinned', originalMessage.rid, '', me, {
-			attachments: [
-				{
-					'text': originalMessage.msg,
-					'author_name': originalMessage.u.username,
-					'author_icon': getAvatarUrlFromUsername(originalMessage.u.username),
-					'ts': originalMessage.ts
-				}
-			]
-		});
+		const attachments = [{
+			'text': originalMessage.msg,
+			'author_name': originalMessage.u.username,
+			'author_icon': getAvatarUrlFromUsername(originalMessage.u.username),
+			'ts': originalMessage.ts
+		}];
+
+		if (originalMessage.attachments !== undefined) {
+			originalMessage.attachments.forEach((attachment) => {
```

Comment:
this loop will merge all message attachments into a single one? this is not what I would expect.